joblib/memory.py

```diff
@@ -131,10 +131,20 @@ def _load_output(output_dir, func_name, timestamp=None, metadata=None,
                     t, str(signature), output_dir))
 
     filename = os.path.join(output_dir, 'output.pkl')
-    if not os.path.isfile(filename):
+
+    n_attempts = 3
+    sleep_time = 0.1
+
+    for _ in range(n_attempts):
```

Comment:
+1 for starting very short and doubling sleep time at each attempt (until we reach a sleep time threshold at which point we raise the error).